syntax = "proto3";

package report;

message ReportRequest {
    string streamId = 1;
    string userId = 2;
    string title = 3;
    string description = 4;
    bool isFrameStored = 5;
}

message ReportResponse {
    string message = 1;
    bool success = 2;
    string result = 3;
}

message ReportStoreRequest {
    string report_id = 1;
}

message ReportStoreResponse {
    string message = 1;
    bool success = 2;
}

service ReportService {
    rpc createReport(ReportRequest) returns (ReportResponse) {};
    rpc saveReport(ReportStoreRequest) returns (ReportStoreResponse) {};
}